### ISSUE TITLE
Problem with reproducing results on CLEVR / CLEVRTEX

### ISSUE THREAD

**Issue by twitwi** [BrunoSauvalle/AST#1]
Thanks for sharing the code of the article.
We tried re-running the experiments, (first) on CLEVR.
We ran it 3 times using the provided configuration.

Each time, after background training, in the segmentation training, the mIoU increases to at most 15% and stays at this value (compared to the 90.2% in the article).

Any idea of what we might need to change to reproduce the results?

**Comment by BrunoSauvalle** [comment]
Hello, 

Please do not consider my previous comment. It seems that the issue is simply caused by a bug in the management of the background complexity flag for simple backgrounds in supervised mode, which is the case for CLEVR. I will soon update the repo to correct this bug.
Thanks for raising this issue and please confirm that the problem is solved with the patch
Regards,

**Comment by SayanChaki** [comment]
Hello,

Thank you for the update you made, having tried with the updated code, we were able to recreate the results for CLEVR. However, for CLEVRTEX on curriculum training, the MIoU does not surpass 36%. Could you please look into that as well?

**Comment by BrunoSauvalle** [comment]
Thank you for the feedback, 

In order to see where is the problem, could you please look at the background dataset you generated during step 2 and indicates whether your generated backgrounds and masks are similar to those appearing p3251 in the WACV AE-NE paper ?  (https://openaccess.thecvf.com/content/WACV2023/papers/Sauvalle_Autoencoder-Based_Background_Reconstruction_and_Foreground_Segmentation_With_Background_Noise_Estimation_WACV_2023_paper.pdf).

Regards,

**Comment by BrunoSauvalle** [comment]
Hello,

Could you please check whether the path you have put in the field "train_dataset_input_path" in the MF_config file is the path to the dataset images lying in the background dataset ( folder "input_images_train") and generated by the command "generate_background_dataset.py" and not to the original dataset . Although these images are the same as in the original dataset, the order of the images has to be the same as in the same as in the background images (folder "backgrounds_rgba_train) located   in  path " “train_dataset_background_path”, which is not the case with the initial dataset.

Regards,

**Comment by BrunoSauvalle** [comment]
Hello,

Thanks for the feedback,

Could you please send me some samples of images generated during training (stored in the directory defined by the path  "training_images_output_directory"  of MF_config config file ), for example at epoch 0,4,8,20,60 ?
For example, training images at epoch 20 should look like the following : 

![comparison_train_set_20](https://github.com/BrunoSauvalle/AST/assets/44267731/c2fd55f2-613c-468b-a9b6-afcb4194c6e1)


Regards,

**Comment by BrunoSauvalle** [comment]
and at epoch 8, you should have something like this : 
![comparison_train_set_8](https://github.com/BrunoSauvalle/AST/assets/44267731/f5c63f62-3310-46c7-8536-18af2f9fc9fc)

**Comment by BrunoSauvalle** [comment]
and the images generated during
![Uploading comparison_train_set_0.png…]()
 epoch zero allows to check the quality of the background model:

**Comment by BrunoSauvalle** [comment]
And the images generated during epoch zero allow to check the quality of the background model : 

![comparison_train_set_0](https://github.com/BrunoSauvalle/AST/assets/44267731/6c73a87b-5bad-4bf5-94a1-9c30daf01274)

**Comment by BrunoSauvalle** [comment]
You may also check that the file paths are updated in the MF_config.py file in a way which is consistent with the github template. For example, the key "train_dataset_background_path" should be associated to the path of the rgba image folder generated by the background model, not the rgb image folder.

**Comment by SayanChaki** [comment]
> You may also check that the file paths are updated in the MF_config.py file in a way which is consistent with the github template. For example, the key "train_dataset_background_path" should be associated to the path of the rgba image folder generated by the background model, not the rgb image folder.

Hu Bruno, thank you for your message. Having considered all the model works appropriately, but I still fail to reach the accuracy with the necessary variations as proposed by you in the paper i.e. for CLEVRTEX 79.58 ± 0.54 and for CLVER 90.27 ± 0.20.

For CLEVR we get close to 87 and CLEVRTEX close to 72.
